lib/matplotlib/_mathtext.py

```diff
@@ -2222,6 +2256,16 @@ def subsuper(self, s, loc, toks):
             result = Hlist([vlist])
             return [result]
 
+        # Set the minimum shifts for the superscript and subscript (node756).
+        constants = _get_font_constant_set(state)
+        if isinstance(nucleus, Char):
+            shift_up = 0
+            shift_down = 0
+        else:
+            drop_amount = constants.subdrop * xHeight * SHRINK_FACTOR
+            shift_up = nucleus.height - drop_amount
+            shift_down = nucleus.depth + drop_amount
```

Comment:
The problem comes from here. `{xyz}^k` triggers the `else` branch, while `xyz^k` triggers the `if` branch. This is consistent with node 756 of the book. However, TeX produces the same results for both. This is likely the result of extremely careful choices of font constants by Knuth. I can patch this by deleting the `if` branch, which should ensure that `{xyz}^k` and `xyz^k` get rendered identically.